src/main/java/org/spongepowered/asm/mixin/MixinEnvironment.java

```diff
@@ -659,8 +662,8 @@ boolean isSupported() {
         /**
          * Java 12 and above
          */
-        JAVA_12(12, Opcodes.V12, LanguageFeature.METHODS_IN_INTERFACES | LanguageFeature.PRIVATE_METHODS_IN_INTERFACES
-                | LanguageFeature.NESTING | LanguageFeature.DYNAMIC_CONSTANTS) {
+        JAVA_12(12, Opcodes.V12, LanguageFeatures.METHODS_IN_INTERFACES | LanguageFeatures.PRIVATE_METHODS_IN_INTERFACES
+                | LanguageFeatures.NESTING | LanguageFeatures.DYNAMIC_CONSTANTS) {
```

Comment:
Should these enums be removed to be consistent with upstream?
[Mumfrey has explained what `CompatibilityLevel` is in forgecord.](https://discord.com/channels/313125603924639766/733055378371117127/853257030416072724)
![image](https://user-images.githubusercontent.com/41327811/123545814-9d563d00-d78c-11eb-8e63-c2265045f949.png)

Reply:
Since this is a mirroring, I would expect removing this when and only when sponge does so as this one doesn't really have particular importance to us.

Reply:
Notice that `JAVA_12` to `JAVA_17` were added by fabric, not sponge, that is, these enums should not exist originally.

Reply:
Then this would probably depend on modmuss to decide.

Reply:
For now we are going to keep the enum values that we have added to ensure that existing mods continue to work, I think it would be confusing for people to target their mixins as JAVA_11 when compling against 16. 

This also keeps the door open to make use of them down the line as chanaces are at somepoint they will be needed anyway.

The bigger question is are they really needed? Could we get away with just using the class version that the mixin was compiled with as the compat level? I might be missing something but I dont really see the need for it.

Reply:
I suspect the logic is running on older Mixin versions which didn't do that it would hit bytecode it didn't know what to do with and explode, otherwise you'd probably be better trusting the version a Mixin was compiled as vs a promise from a string (which can be wrong and lead people into confusing crashes why it thinks it can only support `JAVA_8`.